OpenRA.Mods.Common/Scripting/Properties/TransformProperties.cs

```diff
@@ -28,7 +28,7 @@ public TransformProperties(ScriptContext context, Actor self)
 
 		[ScriptActorPropertyActivity]
 		[Desc("Queue a new transformation.")]
-		public void Deploy()
+		public void Transform()
```

Comment:
If a script field contains functions with the same name and parameters, the game will **crash**. This is the first reason.

Moreover, we now have many traits related with "deploy" actions, if we pick some of them to make script properties, they all have `Deploy()`: 
1. MinelayerProperties, `Deploy()` = lay single mine
2. TransformProperties, `Deploy()` = transform
3. MadTankProperties,  `Deploy()` = demolition
4. GrantConditionOnDeployProperties, `Deploy()` = Get condition
5. GrantConditionOnDeployWithChargeProperties, `Deploy()` = Get timed condition

Since we cannot name it the same,  then using the general name `Deploy()` for any script function is questionable. I would recommend to use a more precise name in this case.

Reply:
```
OpenRA engine version 725f926b29e4b80b5572431c86c5a4e8f6c87a26
OpenRA Language: en
Shattered Paradise mod version {DEV_VERSION}
Date: 2023-09-28 02:08:35Z
Operating System: Windows (X64, Microsoft Windows NT 10.0.19045.0)
Runtime Version: .NET CLR 6.0.11
Installed Language: zh (Installed) zh (Current) zh (Current UI)
Exception of type `Eluant.LuaException`: Actor 'scrglyder2' defines the command 'Deploy' on multiple traits
   at OpenRA.Scripting.ScriptObjectWrapper.Bind(Object[] clrObjects) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Scripting\ScriptObjectWrapper.cs:line 61
   at OpenRA.Scripting.ScriptActorInterface.InitializeBindings() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Scripting\ScriptActorInterface.cs:line 46
   at OpenRA.Scripting.ScriptActorInterface..ctor(ScriptContext context, Actor actor) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Scripting\ScriptActorInterface.cs:line 35
   at OpenRA.Actor.<>c__DisplayClass107_0.<OnScriptBind>b__0() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Actor.cs:line 603
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location ---
   at System.Lazy`1.CreateValue()
   at OpenRA.Actor.<Dispose>b__90_0(World w) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Actor.cs:line 411
   at OpenRA.World.Dispose() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\World.cs:line 603
   at OpenRA.Graphics.WorldRenderer.Dispose() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Graphics\WorldRenderer.cs:line 437
   at OpenRA.Game.StartGame(String mapUID, WorldType type) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 185
   at OpenRA.Network.UnitOrders.ProcessOrder(OrderManager orderManager, World world, Int32 clientId, Order order) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Network\UnitOrders.cs:line 178
   at OpenRA.Network.OrderManager.ReceiveImmediateOrders(Int32 clientId, OrderPacket orders) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Network\OrderManager.cs:line 177
   at OpenRA.Network.NetworkConnection.OpenRA.Network.IConnection.Receive(OrderManager orderManager) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Network\Connection.cs:line 350
   at OpenRA.Sync.<>c__DisplayClass13_0.<RunUnsynced>b__0() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Sync.cs:line 167
   at OpenRA.Sync.RunUnsynced[T](Boolean checkSyncHash, World world, Func`1 fn) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Sync.cs:line 203
   at OpenRA.Sync.RunUnsynced(World world, Action fn) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Sync.cs:line 168
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 631
   at OpenRA.Game.LogicTick() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 646
   at OpenRA.Game.Loop() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 815
   at OpenRA.Game.Run() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 868
   at OpenRA.Game.InitializeAndRun(String[] args) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 305
   at OpenRA.Launcher.Program.Main(String[] args) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Launcher\Program.cs:line 32
```

Reply:
perhaps add an obsolete tag instead

```csharp
[Obsolete("This function will be removed in future versions. Use Transform instead.")]
[ScriptActorPropertyActivity]
[Desc("Queue a new transformation.")]
public void Deploy()
{
	transforms.DeployTransform(true);
}
```

it's in Systems namespace